cloud-builders/helm/helm.bash

```diff
@@ -0,0 +1,102 @@
+  CLOUDSDK_CONTAINER_CLUSTER=<cluster name>
+EOF
+        exit 1
+    }
+
+    [[ -z "$cluster" ]] && var_usage
+    [ ! "$zone" -o "$region" ] && var_usage
+
+    if [ -n "$region" ]; then
+      echo "Running: gcloud container clusters get-credentials --project=\"$project\" --region=\"$region\" \"$cluster\""
+      gcloud container clusters get-credentials --project="$project" --region="$region" "$cluster"
+    else
+      echo "Running: gcloud container clusters get-credentials --project=\"$project\" --zone=\"$zone\" \"$cluster\""
+      gcloud container clusters get-credentials --project="$project" --zone="$zone" "$cluster"
+    fi
+fi
+
+# if HELM_VERSION starts with v2, initialize Helm
+if [[ $HELM_VERSION =~ ^v2 ]]; then
+  echo "Running: helm init --client-only"
+  helm init --client-only
+else
+  echo "Skipped 'helm init --client-only' because not v2"
+fi
+
+# if GCS_PLUGIN_VERSION is set, install the plugin
+if [[ -n $GCS_PLUGIN_VERSION ]]; then
+  echo "Installing helm GCS plugin version $GCS_PLUGIN_VERSION "
+  helm plugin install https://github.com/nouney/helm-gcs --version $GCS_PLUGIN_VERSION
+fi
+
+# if DIFF_PLUGIN_VERSION is set, install the plugin
+if [[ -n $DIFF_PLUGIN_VERSION ]]; then
+  echo "Installing helm DIFF plugin version $DIFF_PLUGIN_VERSION "
+  helm plugin install https://github.com/databus23/helm-diff --version $DIFF_PLUGIN_VERSION
+fi
+
+# if HELMFILE_VERSION is set, install Helmfile
+if [[ -n $HELMFILE_VERSION ]]; then
+  echo "Installing Helmfile version $HELMFILE_VERSION "
+  curl -SsL https://github.com/roboll/helmfile/releases/download/$HELMFILE_VERSION/helmfile_linux_amd64 > helmfile
+  chmod 700 helmfile
+fi
+
+# check if repo values provided then add that repo
+if [[ -n $HELM_REPO_NAME && -n $HELM_REPO_URL ]]; then
+  echo "Adding chart helm repo $HELM_REPO_URL"
+  helm repo add $HELM_REPO_NAME $HELM_REPO_URL
+fi
+
+echo "Running: helm repo update"
+helm repo list && helm repo update || true
+
+
+# if 'TILLERLESS=true' is set, run a local tiller server with the secret backend
+# see also https://github.com/helm/helm/blob/master/docs/securing_installation.md#running-tiller-locally
+if [ "$TILLERLESS" = true ]; then
+  if [[ $HELM_VERSION =~ ^v2 ]]; then
+
+    # create tiller-namespace if it doesn't exist (helm --init would usually do this with server-side tiller'
+    if [[ -n $TILLER_NAMESPACE ]]; then
+      echo "Creating tiller namespace $TILLER_NAMESPACE"
+      kubectl get namespace $TILLER_NAMESPACE || kubectl create namespace $TILLER_NAMESPACE
+    fi
+
+    echo "Starting local tiller server"
+    #default inherits --listen localhost:44134 and TILLER_NAMESPACE
+    #use the secret driver by default
+    tiller --storage=secret &
+    export HELM_HOST=localhost:44134
+    if [ "$DEBUG" = true ]; then
+        echo "Running: helm $@"
+    fi
+    helm "$@" && exitCode=$? || exitCode=$?
+    echo "Stopping local tiller server"
+    pkill tiller
+    exit $exitCode
```

Comment:
I think we can remove this part since Tiller was depreciated 4 years ago when v3.0 was released.

Reply:
ok but you think we should keep the rest of the script in here as the entrypoint?